sensor_test.py

```diff
@@ -0,0 +1,172 @@
+
+    sensor_reading = get_sensor_reading()
+    assert isinstance(
+        sensor_reading, float
+    ), "Sensor doesn't seem to register temperature"
+
+    print("Sanity test passed")
+
+
+def test_reboot(get_sensor_info, reboot_sensor):
+    """
+    Steps:
+        1. Get original sensor info.
+        2. Reboot sensor.
+        3. Wait for sensor to come back online.
+        4. Get current sensor info.
+        5. Validate that info from Step 1 is equal to info from Step 4.
+    """
+    print("Get original sensor info")
+    sensor_info_before_reboot = get_sensor_info()
+
+    print("Reboot sensor")
+    reboot_response = reboot_sensor()
+    assert reboot_response == "rebooting", "Sensor did not return proper text in response to reboot request"
+
+    print("Wait for sensor to come back online")
+    sensor_info_after_reboot = wait(
+        func=get_sensor_info,
+        condition=lambda x: isinstance(x, dict),
+        tries=10,
+        timeout=1,
+    )
+
+    print("Validate that info from Step 1 is equal to info from Step 4")
+    assert sensor_info_before_reboot == sensor_info_after_reboot, ("Sensor info after reboot "
+                                                                   "doesn't match sensor info before reboot")
+
+
+def test_set_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    print("Step 1", 'Set sensor name to "new_name"')
+    updated_sensor_name = "new_name"
+    set_sensor_name(updated_sensor_name)
+
+    print("Step 2", 'Get sensor_info')
+    sensor_info_after_name_change = get_sensor_info()
+
+    print("Step 3", 'Validate that current sensor name matches the name set in Step 1')
+    assert sensor_info_after_name_change.get("name") == updated_sensor_name
+
+
+def test_set_sensor_reading_interval(
+        get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("Step 1", "Set sensor reading interval to 1")
+    new_sensor_reading_interval = 1
+    set_sensor_reading_interval(new_sensor_reading_interval)
+
+    print("Step 2", "Get sensor info")
+    sensor_info_step2 = get_sensor_info()
+
+    print("Step 3", "Validate that sensor reading interval is set to interval from Step 1")
+    assert sensor_info_step2.get("reading_interval") == new_sensor_reading_interval
+
+    print("Step 4", "Get sensor reading")
+    sensor_reading_step4 = get_sensor_reading()
+
+    print("Step 5", "Wait for interval specified in Step 1")
+    wait(func=get_sensor_reading,
```

Comment:
This could be a plain "hard" sleep (`time.sleep(1)`) because the test scenario requires us to wait for exactly the specified amount of time in order to get a fresh reading.

If you decide to keep the conditional wait here, you can combine steps 5 and 6 into one by doing this:
```
sensor_reading_after_wait = wait(...)
```

Reply:
import time was added
time.sleep(new_sensor_reading_interval) used instead of the old way